ts4.1/index.d.ts

```diff
@@ -327,7 +327,7 @@ export function withTranslation<N extends Namespace = DefaultNamespace>(
     withRef?: boolean;
   },
 ): <
-  C extends React.ComponentType<React.ComponentProps<C> & WithTranslationProps>,
```

Comment:
The constrain should not rely on its on generics, this breaks on typescript 4.6 as `C` becomes `unknown` rather than any in the first instance